test/e2e/e2e_test.go

```diff
@@ -292,6 +297,140 @@ func pollForConfigMapChange(client *kubernetes.Clientset, compareConfigMap *v1.C
 	})
 }
 
+// GenerateRoute creates a OpenShift routes
+func GenerateRoute(routeClient *routeclient.RouteV1Client, s *v1.Service, namespace string) (*routev1.Route, error) {
+	route, err := routeClient.Routes(namespace).Create(&routev1.Route{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      s.ObjectMeta.Name,
+			Namespace: s.ObjectMeta.Namespace,
+			Labels:    s.ObjectMeta.Labels,
```

Comment:
I think you can access all the attributes directly from the `s` struct, although generally these attributes of service and route are not logically connected. Good enough for test though.